src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -28,28 +28,29 @@ internal partial class Binder
         /// </summary>
         internal Binder(CSharpCompilation compilation)
         {
-            Debug.Assert(compilation != null);
+            RoslynDebug.Assert(compilation != null);
+            RoslynDebug.Assert(this is BuckStopsHereBinder);
```

Comment:
This seems like an odd requirement to put in a constructor.

Reply:
It's the only thing allowed to have a null _next.